### DISCUSSION TITLE
Could not find an descriptor for expression type \"JavaScript\".

### DISCUSSION THREAD

**Discussion by RinkuKushwaha** [elsa-workflows/elsa-core#5229] [Q&A]
I have added a switch activity in my workflow. and added javascript condition where i am checking one property of my input that i have added dynamically during workflow execution via one activity. In the workflow result i am getting this message **Could not find an descriptor for expression type \"JavaScript\".** as an incident.

Please let me know what is the right way to use switch in elsa 3.0 workflow designer

**Comment by glime-ai** [comment]
To use a JavaScript expression in a switch activity within Elsa 3.0, you need to ensure that the JavaScript expression evaluator is properly configured and that the JavaScript expression is correctly formatted.

Here are some steps to troubleshoot and resolve the issue:

1. **Check JavaScript Evaluator Configuration**: Make sure that the JavaScript expression evaluator is registered and configured in your Elsa setup. This is typically done in the startup configuration of your application.

2. **Correct Expression Syntax**: Ensure that the JavaScript expression you are using in the switch activity is correctly formatted. According to the documentation, you can access workflow variables using the `getVariableName()` convention. Make sure your expression follows this pattern and that the variable you are trying to access exists.

3. **Check for Known Issues**: Look for any known issues or bugs related to the switch activity and JavaScript expressions in the Elsa GitHub repository or the official documentation. There might be updates or workarounds available.

4. **Review Documentation**: Review the official Elsa documentation for examples and guidance on how to use switch activities with JavaScript expressions. The documentation might have specific examples that can help you.

5. **Check for Updates**: Ensure that you are using the latest version of Elsa, as there might have been fixes or improvements related to the issue you are experiencing.

6. **Seek Support**: If you continue to face issues, consider reaching out to the Elsa community or support channels for assistance. You can post a question on the Elsa GitHub discussions or Stack Overflow with specific details about your workflow and the expression you are using.

If you provide the exact JavaScript expression you are using in the switch activity and more details about how you are adding the dynamic input, I can give you more targeted advice.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
